openlibrary/plugins/worksearch/code.py

```diff
@@ -287,14 +287,15 @@ def do_search(
     :param sort: csv sort ordering
     :param spellcheck_count: Not really used; should probably drop
     """
+
+    if web.cookies(sfw="").sfw == 'yes':
+        fields = list(
+            WorkSearchScheme.default_fetched_fields | {'editions'} | {'subject_key'}
```

Comment:
@jimchamp, here we pass in a special directive to `solr` to get the `subject_key` of editions. And based on this, we can see if that book has a content warning and can then blur the image if the patron has "safe for work" mode enabled in their cookies.